### ISSUE TITLE
New domains could come with a better mail config by default

### ISSUE THREAD

**Issue by patch0** [BytemarkHosting/symbiosis#35]
When I'm setting up a new domain, it comes with a default mail setup which currently excludes a number of important-for-deliverability features, such as DKIM and SPF. 

People who have read the documentation will notice this and add the flags, but people who don't, won't. 

It'd be nice if the set-up-a-new-domain cronjob could touch various config items with default values ("default" in the dkim file, generate a dkim.key, touch spf, etc) if the config/ directory doesn't exist yet. 

On a similar note, if symbiosis-xmpp is installed, the "xmpp" file could be added by default at the same time.

Obviously, users would have to have their "rm" of said file respected if they decided to change the default.

> Originally reported on [Bytemark's Gitlab](https://gitlab.bytemark.co.uk/open-source/symbiosis/issues/14) by @patch0 on 2015-10-19T14:35:17.000Z

**Comment by patch0** [comment]
also: antivirus, antispam, and *maybe* the DNSBLs could be activated by default in the same manner.

> Originally posted by @patch0 on 2015-10-22T17:00:22.000Z

**Comment by patch0** [comment]
We could use incrond to watch the new directories as they're created, and run a "skeleton" setup against each one, copying from a directory in `/etc/symbiosis`.

**Comment by ianeiloart** [comment]
Please be careful not to active ClamAV unnecessarily. And maybe even to check the available RAM before doing so. But yeah, DNSBLs - activate them.

**Comment by hairy-dog** [comment]
How about having a "default setup" template that the Admin user could modify?
What I do is keep just that on my laptop, which I copy into place when setting up a new site. 
It would be good if there was a "dummy" default folder installed by Symbiosis that was used as the basis of newly created sites/mailboxes/whatever. 
The user could modify this and it would then be used as the defaults when adding any new domain directories in /srv, or when adding new mailboxes within an existing domain.

**Comment by telyn** [comment]
Yeah I like the idea of having a customisable template, and the presence of a `config/` dir is a good enough sign that the domain has already been configured.

a dir in `/etc/symbiosis` makes sense to me because it would allow more advanced users to SSH in and modify it, prevent more mistake-prone users from accidentally trashing it during their normal workflow, and allow customisations to persist / be modified during dist-upgrades.

It wouldn't prevent @hairydogltd's approach from working, either.

**Comment by patch0** [comment]
Sounds good.  I'd propose `/etc/symbiosis/skel` as the place for this? (Following in the footsteps of `/etc/skel`.)

**Comment by patch0** [comment]
Need to be careful for race-conditions -- `config/dns` is created by `symbiosis-dns-generate` hourly too.  So maybe looking at the `ctime` on the config directory would make sense.

**Comment by patch0** [comment]
Actually the proper fix is to have a hooks directory (like symbiosis-ssl does now) and execute those at the same time.  The symbiosis-dns package would then symlink in symbiosis-dns-configure to this hooks directory, which then gets executed on new domains.

(For the record directories don't appear to have a useful "created at" time, so there's nothing to check there.)

**Comment by telyn** [comment]
Symbiosis-dns-generate would still need to be run on a cron anyway to pick up user changes though, right?

**Comment by telyn** [comment]
Also - I think the skeleton directory should be a 'full' domain, ready to go. I.e. have the public and config folders ready to go. This way we could store the default landing page there too - guaranteeing that we don't end up with situations like we currently see on http://symbiosis-jessie.work.telyn.uk0.bigv.io (this is a freshly-imaged symbiosis vm) - Just a 403 forbidden when there's no index.html

This would also let customers who're doing simple shared-hosting type setups set up their own landing page to give their users some information, or to have people doing multi-site setups get stuff working straight away.

**Comment by patch0** [comment]
Ugh, yes, symbiosis-dns-generate would still need to be run on a cron.  Race-tastic :) 

A full directory would be an interesting prospect!  Let's do it.
